Rewrite readme (#509)
oam-dev/rudr: README.md

![](https://github.com/oam-dev/rudr/workflows/Rust/badge.svg)

Rudr is an [Open Application Model (OAM)](https://github.com/oam-dev/spec/blob/master/1.purpose_and_goals.md) implementation for Kubernetes. 

***Note: Rudr is currently in alpha. It may reflect the API or features we are vetting before inclusion into the OAM spec.***

## Why should I care?

> Kubernetes API resources focused on container infrastructure rather than the applications themselves. Yet users want to think in terms of application architecture, not of infrastructure.

Reviewer: ```suggestion
Kubernetes API resources focused on container infrastructure rather than the applications per se. In contrast, application developers think in terms of application architecture, not of infrastructure.
```

> **Rudr provides application level primitives for Kubernetes that enable:**

Reviewer: ```suggestion
**Rudr provides application level primitives for Kubernetes that enables:**
```

- The ability to define application (e.g., WordPress) in Kubernetes. 
- The ability to define operational capability (e.g., auto-scaling policy, rather than HPA) in Kubernetes.

> - A standard and portable description of Kubernetes application (e.g., a infrastructure agnostic and self-descriptive application definition, includes every dependency and operational ability the application requires to run).  

Reviewer: I think this sentence is overlapped with the first point.
Could you rephrase it to be more focused on portability only?

> - A standard and portable description of Kubernetes application (e.g., a infrastructure agnostic and self-descriptive application definition, includes every dependency and operational ability the application requires to run).  

Reviewer 2: ```suggestion
- The ability to define an infrastructure agnostic and self-descriptive application description, includes every dependency and operational ability the application requires to run.
```

- An application centric abstraction atop container infrastructure.

> **Rudr can be used by:**

Reviewer: I think the "used by" sounds confusing.
Maybe just call it **Rudr use cases** or **user scenarios**.

- Developers who want to describe application from developer's view, rather than learning low level primitives.
- Operators who want to focus on strategies of operating the application, rather than infrastructure details.
- Kubernetes engineers who want to expose application level API to developers and operators, rather than full Kubernetes API.
- Software distributors who want to distribute application to everywhere, regardless of the differences of runtimes.
- Packaging tools, such as Helm and CNAB, to package self-descriptive YAML files of the application as software release
- Platforms to define a serverless application without any infrastructure concepts.

## Get started

Define and deploy a [helloworld-python](./docs/how-to/create_component_from_scratch.md) application with Rudr.

> ## How Rudr works?

Reviewer: ```suggestion
## How does Rudr work?
```
